src/Compilers/VisualBasic/Test/Syntax/Parser/InterpolatedStringParsingTests.vb

```diff
@@ -922,4 +922,22 @@ BC30648: String constants must end with a double quote.
 
     End Sub
 
+    <Fact>
+    Public Sub Test_KeywordExpression()
+        Dim code =
+"Class Example
+  Public ReadOnly Property [End] As Integer
+  Public Overrides Function ToString() As String
+    Return $""{End.ToString()})""
+  End Function
+End Class"
+        Dim result = Parse(code)
+        Dim diagnostics = result.GetDiagnostics.ToImmutableListOrEmpty
+        result.AssertTheseDiagnostics(
+        <expected>
+BC30183: Keyword is not valid as an identifier.
+    Return $"{End.ToString()})"
```

Comment:
can we have a lot more tests than this?  we'll need tests of keywords in many different areas followed by any allowable delimiter.  We should also have specific tree and diagnostics tests

Reply:
this also affects far more than interpolated strings.  i would expect a lot of statement/expression parsing tests.